src/core/server/audit_trail/types.ts

```diff
@@ -19,43 +19,120 @@
 import { KibanaRequest } from '../http';
 
 /**
- * Event to audit.
+ * Audit event schema using ECS format.
+ * https://www.elastic.co/guide/en/ecs/1.5/index.html
  * @public
- *
- * @remarks
- * Not a complete interface.
  */
-export interface AuditableEvent {
+export interface AuditEvent {
+  /**
+   * Human readable message describing the action, outcome and user.
+   *
+   * @example
+   * User 'jdoe' updated saved object 'kpis' of type 'dashboard'
+   */
   message: string;
-  type: string;
+  event: {
+    action: string;
+    category: EventCategory | readonly EventCategory[];
+    type: EventType | readonly EventType[];
+    outcome: EventOutcome;
+    module?: string;
+    dataset?: string;
+  };
+  user: {
+    name: string;
+    email?: string;
+    full_name?: string;
+    hash?: string;
+    roles?: readonly string[];
+  };
+  session?: {
+    id: string;
+  };
+  kibana: {
+    /**
+     * Current space id of the request.
+     */
+    namespace: string;
```

Comment:
nit: `namespace` is a concept which only belongs to saved objects, and is a rather bad abstraction to the space in which the saved object exists. Since this field is not part of `saved_objects`, I think it's more appropriate to call this `space_id`.

I also wonder if we should rename `saved_objects.namespaces` to `saved_objects.space_ids`. While use use namespaces internally, it's not a concept that your average Kibana user is aware of, let alone an auditor. A space is a first-class concept to users though, so that might be easier to understand. What do you (@thomheymann & @jportner) think?

We've also historically tried not to leak X-Pack concerns (such as the concept of a "space") into OSS, so it feels out of place to define this as part of core. I see the original interface used to be defined in `x-pack/plugins/audit_trail/server/types`, but has since been moved here. I'm not sure what the best solution is, but I can see a couple of options:
1) Move this back into X-Pack. This means we can't offer type-safety for OSS though.
2) Extend this interface in X-Pack to add x-pack specific constructs. OSS shouldn't care about them, so removing them from here shouldn't impact them. It may make type-checking in X-Pack more complex though.
3) Ignore my pedantic concerns and leave it as-is

Reply:
I don't see this interface as necessarily x-pack specific as the schema is just describing ECS standard (which I think is open source) and eventually it would be good to use that for the Kibana logging format too. Happy to move this to x-pack though if it belongs there but it's a bit tricky at the moment as parts of audit logging sit in core and parts sit in x-pack. 

We can change namespace fields to space_ids (I have no preference) but the current naming is aligned with alert team so they would need to change it as well. @pmuellr - Thought?

Reply:
> I also wonder if we should rename `saved_objects.namespaces` to `saved_objects.space_ids`. While use use namespaces internally, it's not a concept that your average Kibana user is aware of, let alone an auditor. A space is a first-class concept to users though, so that might be easier to understand.

Makes sense to me.